src/cwt/claim/ietf_token_status.rs

```diff
@@ -0,0 +1,151 @@
+//! Claims from the IETF Token Status List spec.
+
+use super::{Claim, Error as ClaimError, Key};
+use serde::{Deserialize, Serialize};
+use serde_cbor::Value;
+use std::collections::BTreeMap;
+
+/// Status Claim for the ReferencedToken.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct ReferencedTokenStatus {
+    pub status_list: StatusListReference,
+}
+
+/// Claim describing the location of the Referenced Token in the Status List.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct StatusListReference {
+    pub idx: StatusListIndex,
+    pub uri: StatusListUri,
+}
+
+/// Index to check for status information in the Status List for the Referenced Token.
+#[derive(Copy, Clone, Debug, Deserialize, Serialize, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct StatusListIndex(pub u32);
+
+/// The URI where one can fetch the Status List or Status List Token
+/// containing the status information about the ReferencedToken.
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, PartialOrd, Eq, Ord, Hash)]
+pub struct StatusListUri(pub String);
+
+impl ReferencedTokenStatus {
+    pub fn new(idx: StatusListIndex, uri: StatusListUri) -> Self {
+        Self {
+            status_list: StatusListReference { idx, uri },
+        }
+    }
+}
+
+impl Claim for ReferencedTokenStatus {
+    fn key() -> Key {
+        Key::Integer(65535)
+    }
+}
+impl From<ReferencedTokenStatus> for Value {
+    fn from(value: ReferencedTokenStatus) -> Self {
+        Value::Map(value.into())
+    }
+}
+impl From<ReferencedTokenStatus> for BTreeMap<Value, Value> {
+    fn from(value: ReferencedTokenStatus) -> Self {
+        BTreeMap::from_iter([(StatusListReference::key().into(), value.status_list.into())])
+    }
+}
+impl TryFrom<Value> for ReferencedTokenStatus {
+    type Error = ClaimError;
+
+    fn try_from(value: Value) -> Result<Self, Self::Error> {
+        Ok(serde_cbor::value::from_value(value)?)
+    }
+}
+
+impl Claim for StatusListReference {
+    fn key() -> Key {
+        Key::Text("status_list".into())
+    }
+}
+impl From<StatusListReference> for Value {
```

Comment:
If my comment above is right, I think you can get rid of the conversion impls and just rely on `serde` to serialize/deserialize the `ReferenceTokenValue`.